sdk/python/feast/infra/aws_dynamodb_provider.py

```diff
@@ -0,0 +1,232 @@
+import os
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import boto3
+import mmh3
+import pandas
+from botocore.exceptions import ClientError
```

Comment:
```suggestion
try:
    import boto3
    from botocore.exceptions import ClientError
except ImportError as e:
    from feast.errors import FeastExtrasDependencyImportError
    raise FeastExtrasDependencyImportError("aws", str(e))
```
perhaps adapt this pattern https://github.com/feast-dev/feast/blob/73ac3de613a729da41b7c832bb4a520132399339/sdk/python/feast/infra/gcp.py#L29-L35

Reply:
wow @tedhtchang, you're on the money :)